src/marqo/core/inference/models/abstract_model.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+
+class AbstractModel(ABC):
```

Comment:
I don't have a good alternative to suggest, but model is confusing here because elsewhere we refer to pydantic models

Reply:
`AbstractEmbeddingModel`

Reply:
done